overlord/assertstate/validation_set_tracking_test.go

```diff
@@ -157,3 +186,71 @@ func (s *validationSetTrackingSuite) TestGet(c *C) {
 	err = assertstate.GetValidationSet(s.st, "foo", "baz", &res)
 	c.Assert(err, Equals, state.ErrNoState)
 }
+
+func (s *validationSetTrackingSuite) mockAssert(c *C, name, sequence, presence string) asserts.Assertion {
+	snaps := []interface{}{map[string]interface{}{
+		"id":       "yOqKhntON3vR7kwEbVPsILm7bUViPDzz",
+		"name":     "snap-b",
+		"presence": presence,
+	}}
+	headers := map[string]interface{}{
+		"authority-id": s.dev1acct.AccountID(),
+		"account-id":   s.dev1acct.AccountID(),
+		"name":         name,
+		"series":       "16",
+		"sequence":     sequence,
+		"revision":     "5",
+		"timestamp":    "2030-11-06T09:16:26Z",
+		"snaps":        snaps,
+	}
+	as, err := s.dev1Signing.Sign(asserts.ValidationSetType, headers, nil, "")
+	c.Assert(err, IsNil)
+	return as
+}
+
+func (s *validationSetTrackingSuite) TestEnforcedValidationSets(c *C) {
+	s.st.Lock()
+	defer s.st.Unlock()
+
+	tr := assertstate.ValidationSetTracking{
+		AccountID: s.dev1acct.AccountID(),
+		Name:      "foo",
+		Mode:      assertstate.Enforce,
+		Current:   2,
+	}
+	assertstate.UpdateValidationSet(s.st, &tr)
+
+	tr = assertstate.ValidationSetTracking{
+		AccountID: s.dev1acct.AccountID(),
+		Name:      "bar",
+		Mode:      assertstate.Enforce,
+		PinnedAt:  1,
+		Current:   3,
+	}
+	assertstate.UpdateValidationSet(s.st, &tr)
+
+	tr = assertstate.ValidationSetTracking{
+		AccountID: s.dev1acct.AccountID(),
+		Name:      "baz",
+		Mode:      assertstate.Monitor,
+		Current:   5,
+	}
+	assertstate.UpdateValidationSet(s.st, &tr)
+
+	vs1 := s.mockAssert(c, "foo", "2", "required")
+	c.Assert(assertstate.Add(s.st, vs1), IsNil)
+
+	vs2 := s.mockAssert(c, "bar", "1", "invalid")
+	c.Assert(assertstate.Add(s.st, vs2), IsNil)
+
+	vs3 := s.mockAssert(c, "baz", "5", "invalid")
+	c.Assert(assertstate.Add(s.st, vs3), IsNil)
+
+	valsets, err := assertstate.EnforcedValidationSets(s.st)
+	c.Assert(err, IsNil)
+
+	// foo and bar are in conflict, use this as an indirect way of checking
+	// that both were added to valsets.
```

Comment:
this works but is not very realistic, we should probably leave a TODO/XXX to switch to a test using the Check* methods once they have landed